### DISCUSSION TITLE
Developer Onboarding Program via Zealy

### DISCUSSION THREAD

**Discussion by piber-dev** [akash-network/community#457] [Governance Proposals]
# Developer Onboarding Funding Proposal

## Introduction

After believing that there was a lack of community programs on Akash, discussion led to start one which eventually led to the creation of the Zealy working group. Zealy is a score-based platform where users earn points from doing quests that we want to use for educational, marketing, and innovative reasons. We want to utilize this platform to onboard and educate developers by incentivizing them and leading them to great resources. After reaching out to Tyler and Scott, we were able to run a successful pilot with funding from the community support proposal showing the rough idea on what can be done with Zealy and the results were: 925 AKT rewarded between 225 completed missions and 76 users ([detailed info here](https://docs.google.com/spreadsheets/d/1IKAEcF5rZheg7QK_sClkCE7DUoK93tjeEdTC89Yt6Hw/edit#gid=477009662)).

Community members and Vanguards piberdev and robertdelrey have been pushing the initiative since October seeking funding and successfully launching two pilot previews of what the program could look like. During this period a lot has been learned: e.g. optimizations have been made in user retention by having auto-approval on the first mission; common abuse on the program has been found and dealt with; and more. By implementing all learnings, even more success is to be expected.

> Note: this will be discussed during the Steering Committee call on Thursday (2024-01-25) and the details of this proposal including rewards, payments, and expected results have all been agreed to be reasonable according to the wg-zealy call today (2024-01-22).

## Resources required

1x community manager (taking the lead on wg-zealy calls; writing announcements; paying out rewards; interacting & making final decisions, etc)
1x administrator (reviewing missions, giving technical support, creating helper scripts, finalizing details, etc)
(+ a common collaborative effort on creating tasks)

We believe that each role will approximately need to work a total of 5 hours per week if the program does not scale away from Zealy:
> 1 hour/week/user for wg-zealy calls
> 4 hours/week/user for support, announcements, reaching out, reviewing, etc depending on user role

## Expected outcomes

- Successful implementation of "Zealy Developer Roles" on Discord or "Zealy Developer Certificates" where users can get access to a private channel on Discord and have a colored name and badge on Discord.
- 250-1000 unique users partaking in missions learning and earning from the different missions. This is about 3x or more than the pilots achieved but all learnings and ideas will help adapt the program to achieve these goals.
- A big number of traffic to other Akash-related topics such as: the Insiders program, Akash EDU, the bounties page, different WGs/SIGs and more. Missions will be created to encourage the participation of these other Akash activities but will be carefully thought over as not to bring spam into these topics.
- A clear path for users will be created so that they can learn how to use Akash with minimal technical experience, and get enough AKT to get started whilst doing so. The hope is that users will stick around and remain active for longer than just the bare minimum that the missions are requesting.

## Proposal funding

Pricing at $50 USD/hr, we are requesting $250 USD/week/user for a quarter for a total of $6500 USD for role payments.

Furthermore, we expect to use an upwards of $2500 USD/month for rewards for a total of $7500 USD.

All in all, this brings the requested funding to a total of $14000 USD worth of AKT at the time of submitting the proposal.

Previous left-over funding from the other proposal will be returned to the community pool wallet before submitting the proposal to chain.

After the quarter this proposal is targeted for, there will be a one-month long deadline to pay back any remaining funds to the community pool.

**Comment by rodri-r** [comment]
Great proposal Piber and Robert

As I mentioned in some of the Zealy calls, I was a little skeptical about this, but after seeing the two pilots, I do see more activity in Discord and in my opinion this program is onboarding new users and developers to the network. 

The expected outcomes look great and I do think that they can be achieved.  

Funding looks good too and I think this will bring a lot of benefit to the network, as well as the other Akash-related topics such as: the Insiders program, Akash EDU, the bounties page, different WGs/SIGs, the Akashathons and more.

Keep up the great work guys, congrats!

**Comment by CaptainSplince** [comment]
Thanks for this proposal Piber. Very well thought out. Couple of questions.

- The ([detailed info here](https://docs.google.com/spreadsheets/d/1IKAEcF5rZheg7QK_sClkCE7DUoK93tjeEdTC89Yt6Hw/edit#gid=477009662)) section is a little hard to read. Do you have a breakdown of how many participants completed each Zealy challenge? Also, of the total number of participants, how many completed all the challenges? One other data question, of the total number of Zealy challenges how many are or were considered technical in nature? I'd like to better understand the breakdown between technical participants and non technical participants.

- Do you have any data from Zealy on how many total community members are on the Zealy platform? And of that number how many would be considered developers or technical? 

- One other suggestion is a breakdown of the timeline/roadmap including benchmarks. Example:
-- Week one (tasks and outcomes)
-- Week two (tasks and outcomes)
-- Week three (tasks and outcomes)
-- Week four (WG call and update on progress)

- One other suggestion is creating the list of proposed Zealy quests and the corresponding AKT amount given per quest. That way you can get some feedback on the quests and the AKT amount given. Also, if the propsoal passes you can hit the ground running with creation of said quests.

- One final thought. What will be the user journey? Meaning, how will you track and measure new recruits into the Akash community? Will it be new Discord members, newsletter signups, Twitter/X followers, Github contributors, ect. Completing quests is great, but how will you encourage people who complete quests to stick around and become full fledges Akash community members?

Thanks again. Stoked for this proposal!

**Comment by piber-dev** [comment]
hello @CaptainSplince, thanks for your response.

- you can see the breakdown for how many missions were completed below. from these, the highlighted ones are somewhat technical to technical where you can be the judge on how technical they are. no user completed all missions and you can refer to the link you have in your response for the number of missions completed for every user.

> Check out Akash Bounties! : 28
**Launch a website on Akash Network in 2 minutes!! : 7**
**Super cloud AI Revolution: 4**
**Deploy Wordpress website : 4**
**Create your own SDL using Cloudmos: 17**
**Deploy Pac-Man on Akash: 22**
**Getting started with Cloudmos: 35**
**Install Akash CLI: 31**
Get access to Akash Zealy Community : 76
**Set up an Akash GPU Testnet Provider using Praetor : 1**

- we had 76 members complete the first mission but beyond this we cannot produce data on this point. while some users are sharing their discord handles, not everyone is. @rodri-r has noticed users from zealy sticking around but this is not objective data.

- we plan on doing this in a few different ways. for one planned mission as an example, we are planning to take note of people's docker hub usernames & seeing if they are new developers or not and if they remain active. we can't force users to stick around, but we are planning to try get users involved in different parts of the community including SIGs/WGs and more through missions and referring to various akash-related resources.

thanks for the other suggestions as well! will discuss these suggestions with robert privately to potentially make an update from these.

**Reply by rodri-r** [reply]
Hey @CaptainSplince like Piber says and like I said before in here and in some of our Zealy calls, I was a little skeptical about this pilot at first, not because of Piber or Robert, but because of my experience with Zealy before. 

I have seen lots of scavengers and two footed rats in there, in other communities and that is why I thought this might not work or that some people would take advantage of it.

But after the pilots, I have seen more activity in Discord and it looks like they are coming from this Zealy program.

I think both Piber and Robert have done a great job organizing this and reducing those bad actors as much as possible. 

I like their new goals and how they have set up everything and I am sure that they will keep improving this over time, thus onboarding more users and devs into Akash.

**Comment by DappWiz** [comment]
Thanks @piber-dev for presenting this proposal. After reviewing your proposal, I'd like to highlight that Omniflix is already pursuing a similar educational/onboarding initiative with their Akashedu plan, as described on their website at: https://akashedu.omniflix.tv/
Could you please elaborate on the unique selling proposition (USP) you bring to the table in comparison to Omniflix? Additionally, it would be valuable to understand the impact metrics you aim to achieve.

Moreover, as suggested by @CaptainSplince already, breaking down the tasks is crucial. It would be beneficial if you could provide a detailed breakdown of the tasks, you will be undertaking. Furthermore, consider outlining a timeline or roadmap with benchmarks, such as:

- Week one (tasks and outcomes)
- Week two (tasks and outcomes)
- Week three (tasks and outcomes)
- Week four (Working Group call and progress update)

Your attention to these details will enhance the understanding and evaluation of proposal. Thank you.

**Reply by rodri-r** [reply]
Hey

This Zealy program and Akash Edu compliment each other. This has also been addressed before in here and in our Zealy calls.

Also could you please give some hints as to who you are.

Looks like you just created this Github account 22 minutes ago, you have only posted here and forked a repo to make your account look like it has something.

**Reply by ghost** [reply]
Hey there, this is to complement @piber-dev answers and add more information about your comments and questions 

```
Do you have any data from Zealy on how many total community members are on the Zealy platform? And of that number how many would be considered developers or technical?
```
I got these numbers from the latest Zealy Newsletter (accesible via Zealy official Discord)
New Communities per Blockchain (Updated biweekly) 

- 30 communities on Ethereum
- 9 communities on Binance Smart Chain
- 9 communities on Polygon

New Communities per Vertical (Updated biweekly) 
- 30 defi communities
- 24 gaming communities
- 12 art communities

Sadly it does not offer more info than this 

About how many communities are considered developers or technical in nature, I honestly do not know, its up to each community to decide which way they go

Most of the Zealy missions we’ll create are for developers, though this does not mean we can add 1 or 2 other kind of missions depending on what we want to achieve

 
```
One other suggestion is a breakdown of the timeline/roadmap including benchmarks. Example:
    - - Week one (tasks and outcomes)
    - - Week two (tasks and outcomes)
    - - Week three (tasks and outcomes)
    - - Week four (WG call and update on progress)
```

I’d rather go in a monthly basis tasks and outcomes 
Because its simpler and based on the nature of Akash Zealy missions we have to take time to review missions and grant rewards, our missions arent as easy as "like this tweet"

This is how I would do it in a monthly basis 
For example: 

Month 1

- Tasks: Create 5 developer missions, 1 technical content creation mission, 1 normal social media content mission
- Outcomes: Get 100 submissions, pay XXX much in rewards, recruit 5 new Akash Insiders out of the most winners
- Have 2 WG calls and updates with the community

We tried to go week by week, however, things got messy in regards of mission quality and communications 

```
- One other suggestion is creating the list of proposed Zealy quests and the corresponding AKT amount given per quest. That way you can get some feedback on the quests and the AKT amount given. Also, if the proposal passes you can hit the ground running with creation of said quests.
```

Sure, there are a few missions in draft already created, I can share those in the Zealy discord chat and have them reviewed by the members, what to do in case no ones provides feedback on the prizes? 

```
- One final thought. What will be the user journey? Meaning, how will you track and measure new recruits into the Akash community? Will it be new Discord members, newsletter signups, Twitter/X followers, Github contributors, ect. Completing quests is great, but how will you encourage people who complete quests to stick around and become full fledges Akash community members?
```

This is my favorite question, we can make efforts recruit the best talents from the Zealy initiative and depending on their talents we can route them to the Akash insiders, or get them involved in other working groups

I think we could have a recruitment goal, say, 

Every month recruit: 
3 new Akash insiders  
5 Workers for XX, YY and ZZ working groups 
Have 2 KOLs posts about Akash (with requisites)   

These are just ideas, but what I mostly try to say is that Zealy is very customizable, if we want to bring devs, or content creators, or advocates of Akash, we can create missions for that, its going to come down to how good we can create the missions 

Piber and I brainstomr about ideas all the time, but if anyone has a very specific task we could do to help us achieve our objective to onboard developers to Akash, welcome   

—

Side note, we choose Zealy because I’ve been using it for a while and I know what it can and cant do, plus looks pretty, still if you know of another app that can helps us do the job, I’d be happy to check it out 

About Omniflix Akash EDU comment, I dont see why Akash EDU and Akash Zealy couldnt compliment each other

Example of Zealy mission tailored for Akash EDU: 

1 Go to Akash EDU and complete all the courses for a special Discord role 

2 Create THE BEST video guide about “X thing you can do on Akash network”  for a chance to get featured on Akash EDU 

See how these are not even monetary, we can have missions that dont pay AKT rewards, Missions that pay rewards to all the participants and missions that just pay for the best submission (only one winner) 

Hopefully this answers most questions and lead us to the next steps @CaptainSplince @DappWiz

**Reply by CaptainSplince** [reply]
Thanks @instafinanzas & @rodri-r.

I view AkashEDU and Zealy as complementary as well. 

AkashEDU is still a new initiative, but the long term vision for Akash EDU is to make it a rubust, in-depth professional development program.  AkashEDU aims to give students the tools they need to find a technical job within the DePin arena or launch their own project using Akash technologies. 

I view Zealy as an entry point into the Akash ecosystem. Users who complete quests in Zealy are primed to dive deeper via AkashEDU or otherwise.

**Comment by hewi333** [comment]
Hi there,

I appreciate all the work that is going into this effort as it is important to constantly bring in new folks to the community. Coming out of the working group call today, I think there were a few points that could use some thought.

- This is an onboarding program, how do we determine when someone is "onboarded"?
- What specific metrics of success can we decide to report against at the end of the proposed timeline? To provide data for future iterations.
- To save rewards for onboarding NEW individuals, how can we avoid existing community members from just continuously doing Zealy missions to get AKT on a recurring basis? Some potential ideas as we are brainstorming:
  - Lifetime cap on number of Zealy missions someone can complete (# of sprints / missions / etc.)
  - 3-6 months from first mission submission, no longer eligible for Zealy rewards

Again, I like this program and think it's a great to grow the community. I also know this is a learning experience and can't expect perfection right out of the gate.

**Reply by piber-dev** [reply]
hello @heyhewi - thank you for your response!

- this is in our opinion up to the user when they are fully "onboarded." we want to give out roles for specific sets / types of missions on discord if we can arrange this for being e.g. technical or able to set up a provider. when providing results we want to differentiate between different groups in how many were onboarded and in which categories and likely one user completing the start mission or at least two missions including the start mission will count as one user in the results only that we will present it in groups as well.
- i can think of these metrics from the top of my head: total or monthly AKT, total number of participants (including/excluding current ones), number of participants per sprint, missions completed all in all or in certain categories, average participants in wg calls, number of endorsements from users, potentially bringing forward some power users found from tracking new docker hub accounts, and other data easily obtained via missions and scripts. which ones do you think are the best ones to track, and do you have any suggestions on top of these?
- we support a variant of this idea but it is not decided yet how we want to proceed. we also discussed a cap on some categories of missions which would be great if we still want some sort of "bounty-based missions" only available to the best submission.

**Comment by zJuuu** [comment]
Hey, i really like the initiative and idea to have different channels for onboarding developers.
Zealy can be a interesting part in this because most projects use it for simple engagement tasks. Using it to onboard only developers is something new and has to be evaluated. 
In my view, the results of the first pilot program have not really been evaluated enough to give a real insight in to the potential of onboarding through zealy. The data is not adjusted for people who were already active in the community and we don't know how many of these newly onboarded users are still part of the community or only joined because of the rewards and left after.

If you give me access to all the data from zealy from the past pilot i am open to create a few metrics that could be interesting for the community.

That's why i would like to propose a 4-Month pilot with the 4rd month only for evaluation of the "onboarded" users.
For me this means:
- Clear Goals (In example: In the evaluation period: x% of the new users which joined the community through zealy are part of the akash insiders program or x% of zealy participants are still active in the discord)
- Clear target group(s) (What kind of developers are we focusing on? If we want to aquire new Developers do we need social media tasks or focus on technical tasks? In my opinion have some pre requirements like basic knowledge in docker and focusing on developer tasks could filter out users who only try to do the tasks for the rewards and incentivices user who are really interested and reduces support efforts. If there are several target groups, a clear distinction should be made between)
- Cap on total AKT users can earn. (This can be adjusted after the 3 month pilot because we have data about how much on average a new active community member costs.)

Combined with a yearly general onboarding survey for akash this could lead to a targeted approach which onboarding channels are worth focusing on.

I am optimistic that zealy can be a success to onboard new developers and i am looking forward to it! I am also open to supporting you with the evaluation of each phase.

**Reply by CaptainSplince** [reply]
Great framework @zJuuu. I think the next iteration of an Akash x Zealy collaboration needs to layout clear objectives. Do X to get Y which in turn rewards the Akash ecosystem with Z.

**Reply by piber-dev** [reply]
hello @zJuuu and thank you for your response.

> In my view, the results of the first pilot program have not really been evaluated enough to give a real insight in to the potential of onboarding through zealy. The data is not adjusted for people who were already active in the community and we don't know how many of these newly onboarded users are still part of the community or only joined because of the rewards and left after.
> 
> If you give me access to all the data from zealy from the past pilot i am open to create a few metrics that could be interesting for the community.

this is correct but with what we have tracked and asked for this is not something that is fully trackable. please let us know what data you are requesting because the only one mentioned here is already public via the leaderboard going through their connected accounts. we prefer to keep sensitive information private and public information public for the sake of our participants' privacy. :pray:

to speak on your bulletpoints, we think that tracking how active zealy participants can be important on discord, github, and on docker hub. we do want to target multiple groups, but our main focus will be put towards developers and new developers. besides this we also want users to get involved in the insiders program, bounties, and different WGs/SIGs to name a few examples. lastly, a cap is something we are interested in implementing in some way but we are not certain in which way yet.

**Comment by ghost** [comment]
Alright alright!! 

See some of the upcoming missions for Akash onboarding on Zealy  

**Build on Akash Missions**

- Create a basic web-app in Python & deploy it on Akash 

**Akash content Missions**

- Create Shorts style Akash video
- Create ai-powered Akash Videos or logo animations using SDXL

**WGs mission**

- Become a provider of Akash Network 
- Provide persistent storage

And more cool missions coming soon, remember to engage with the Github discussion before going on chain in a governance proposal 

Github discussion Link: https://github.com/orgs/akash-network/discussions/457

Here are mission ideas that didn't make the cut for this phase but who knows about the future

- Create AI powered Akash wiki to find all the info and documentation about Akash Network
- Create tools for dynamic pricing based on supply and demand
- Create blockchain explorer thats deployed on Akash
- Create a video about how to create a virtual machine and host it on Akash
- Design and build a reputation systems for service providers

**Comment by ghost** [comment]
Hey all!! Proposal is live on chain, make sure to cast your vote no matter what it is!! 

https://www.mintscan.io/akash/proposals/247

**Comment by piber-dev** [comment]
[The proposal has passed!!!](https://www.mintscan.io/akash/proposals/247) Thank you everyone who helped change the proposal with all feedback in comments and messages over the past couple of weeks. Announcements will be made very soon on different social channels and this discussion will be closed. 🎉🎉🎉

**Comment by ghost** [comment]
Zealy campaign wrap up thread up https://github.com/orgs/akash-network/discussions/599
